fdbclient/ManagementAPI.actor.cpp

```diff
@@ -264,6 +281,19 @@ ACTOR Future<ConfigurationResult::Type> changeConfig( Database cx, std::map<std:
 		m[initIdKey.toString()] = g_random->randomUniqueID().toString();
 		if (!isCompleteConfiguration(m))
 			return ConfigurationResult::INCOMPLETE_CONFIGURATION;
+	} else {
+		state Future<DatabaseConfiguration> fConfig = getDatabaseConfiguration(cx);
+		Void _ = wait( success(fConfig) || delay(1.0) );
```

Comment:
Are you including the `delay` here because there are some cases where it will never be possible to read the configuration without first setting it?

Reply:
If you configure "triple" when there is only a single process in the cluster the database will become unavailable and therefore you will not be able to read the database configuration. To resolve this you need to configure "single", so that operation needs to happen even if the database is down.